scripts/utils.js

```diff
@@ -25,11 +25,21 @@ function wmAvailabilityCheck(url, onsuccess, onfail) {
  * @param url {string}
  * @return {bool}
  */
-function isValidSnapshotUrl(url) {
+function isValidUrl(url) {
   return ((typeof url) === "string" &&
     (url.indexOf("http://") === 0 || url.indexOf("https://") === 0));
 }
```

Comment:
Question: Is it possible to crash this function if provided with an invalid unicode string?
Can you add unit tests like that to make sure it doesn't happen?
If it crashes, it would be useful to have exception handling here.

This comment applies to other functions as well.